x-pack/dockerlogbeat/pipelinemanager/config.go

```diff
@@ -0,0 +1,143 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package pipelinemanager
+
+import (
+	"crypto/sha1"
+	"reflect"
+	"sort"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/common/transform/typeconv"
+)
+
+// ContainerOutputConfig has all the options we'll expect from --log-opts
+type ContainerOutputConfig struct {
+	Endpoint        string `struct:"output.elasticsearch.hosts"`
+	User            string `struct:"output.elasticsearch.username"`
+	Password        string `struct:"output.elasticsearch.password"`
+	Index           string `struct:"output.elasticsearch.index"`
+	Pipeline        string `struct:"output.elasticsearch.pipeline"`
+	APIKey          string `struct:"output.elasticsearch.api_key"`
+	Timeout         string `struct:"output.elasticsearch.timeout,omitempty"`
+	BackoffInit     string `struct:"output.elasticsearch.backoff.init,omitempty"`
+	BackoffMax      string `struct:"output.elasticsearch.backoff.max,omitempty"`
+	CloudID         string `struct:"cloud.id"`
+	CloudAuth       string `struct:"cloud.auth"`
+	ProxyURL        string `struct:"output.elasticsearch.proxy_url"`
+	ILMEnabled      bool   `struct:"setup.ilm.enabled"`
+	ILMRollverAlias string `struct:"setup.ilm.rollover_alias"`
+	ILMPatterns     string `struct:"setup.ilm.pattern"`
+	TemplateName    string `struct:"setup.template.name"`
+	TempatePattern  string `struct:"setup.template.pattern"`
+}
```

Comment:
how about using "omitempty" for every string value? Otherwise we will end up with a configuration that has empty object. The final error message would be different in that case.